Library/Homebrew/rubocops/extend/formula_cop.rb

```diff
@@ -0,0 +1,19 @@
+module RuboCop
+  module Cop
+    class FormulaCop < Cop
+      @registry = Cop.registry
+
+      def file_inside_formula_root?
+        return false unless processed_source.path =~ %r{/Taps/homebrew}
+        true
+      end
+
+      def on_class(node)
```

Comment:
RuboCop processes the source code and creates an Abstract Syntax Tree(AST) representation, and traverses the AST. As it encounters a `node` of some `type`, a message is sent to cop instance class hierarchy `send(:"on_#{node.type}", node)`
and we override it by defining `on_class` to check and add violations.